packages/app-project/pages/_app.js

```diff
@@ -78,13 +79,18 @@ function MyApp({ Component, pageProps }) {
       )
     }
 
+    if (showCustom404) {
```

Comment:
You can pass `statusCode=404` in the page props, and take advantage of the existing error block (lines 73 – 80) to show a custom 404 page for individual projects.

https://github.com/zooniverse/front-end-monorepo/blob/677c528b6a2279e71c1241fbab57673d2c5838cd/packages/app-project/pages/_app.js#L73-L80

Reply:
So, when a project doesn't exist, return `notFound: true` to show the default 404 page for the app.

When a project does exist, but a project page doesn't exist for that project, return something like this to use a custom 404 error page for a specific project. The project will be passed down in page props, so you can then pass it into the error page as a prop.

```js
{
  notFound: false,
  props: {
    statusCode: 404,
    //… other page props here
  }
}
```